api/tools/src/main/java/org/pmiops/workbench/tools/CreateExtractionBillingProjectWorkspace.java

```diff
@@ -0,0 +1,133 @@
+
+      BillingApi billingApi = apiClientFactory.billingApi();
+      String billingAccount = opts.getOptionValue(billingAccountOpt.getLongOpt());
+      String billingProjectName = opts.getOptionValue(billingProjectNameOpt.getLongOpt());
+      String workspaceName = opts.getOptionValue(workspaceNameOpt.getLongOpt());
+
+      FirecloudCreateRawlsBillingProjectFullRequest billingProjectRequest =
+          new FirecloudCreateRawlsBillingProjectFullRequest()
+              .billingAccount("billingAccounts/" + billingAccount)
+              .projectName(opts.getOptionValue(billingProjectNameOpt.getLongOpt()));
+
+      log.info("Creating billing project");
+      billingApi.createBillingProjectFull(billingProjectRequest);
+
+      while (billingApi
+              .billingProjectStatus(billingProjectRequest.getProjectName())
+              .getCreationStatus()
+          != FirecloudBillingProjectStatus.CreationStatusEnum.READY) {
+        log.info("Billing project is not ready yet");
+        Thread.sleep(5000);
+      }
+
+      String strippedName = workspaceName.toLowerCase().replaceAll("[^0-9a-z]", "");
+      DbWorkspace.FirecloudWorkspaceId workspaceId =
+          new DbWorkspace.FirecloudWorkspaceId(billingProjectName, strippedName);
+
+      FirecloudWorkspaceIngest workspaceIngest =
+          new FirecloudWorkspaceIngest()
+              .namespace(workspaceId.getWorkspaceNamespace())
+              .name(workspaceId.getWorkspaceName());
+
+      log.info(
+          "Creating workspace with ("
+              + workspaceId.getWorkspaceNamespace()
+              + ", "
+              + workspaceId.getWorkspaceName()
+              + ")");
+      FirecloudWorkspace workspace =
+          apiClientFactory.workspacesApi().createWorkspace(workspaceIngest);
+
+      log.info("Updating Workspace ACL");
+      List<FirecloudWorkspaceACLUpdate> acls =
+          Arrays.stream(opts.getOptionValue(ownersOpt.getLongOpt()).split(","))
+              .map(
+                  email ->
+                      new FirecloudWorkspaceACLUpdate()
+                          .email(email)
+                          .accessLevel(WorkspaceAccessLevel.OWNER.toString())
+                          .canCompute(true)
+                          .canShare(true))
+              .collect(Collectors.toList());
+      apiClientFactory
+          .workspacesApi()
+          .updateWorkspaceACL(workspace.getNamespace(), workspace.getName(), false, acls);
+
+      log.info("Workspace created by " + workspace.getCreatedBy());
+      String proxyGroup = apiClientFactory.profileApi().getProxyGroup(workspace.getCreatedBy());
+      Matcher m = Pattern.compile("PROXY_(\\d+)@.*").matcher(proxyGroup);
+      m.matches();
+      String proxyGroupId = m.group(1);
+      log.info(workspace.getNamespace());
+      String petSaAccount =
+          "pet-" + proxyGroupId + "@" + billingProjectName + ".iam.gserviceaccount.com";
```

Comment:
IIRC this is an implementation detail, and not guaranteed by the system, but looks like it's documented at Terra support so I guess it's safe enough for now: https://support.terra.bio/hc/en-us/articles/360031023592-Pet-service-accounts-and-proxy-groups-#h_01ERQNB7J1WEW2QSSWHM8PDRBZ